docs/DeveloperGuide.md

```diff
@@ -1,38 +1,177 @@
 # Developer Guide
 
-## Acknowledgements
+## Design
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+{To Be Added}
 
-## Design & implementation
+---
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## Implementation
+
+### Add feature
 
+The add feature is implemented using a `AddCommand` class extending from `Command` Class. 
+`AddCommand` contains an `execute` methods, which utilises the internal `spiltDetails` methods to extract the food details
+and to construct new `food` object .
+It is also facilitated by `Foodlist` Class, using `addFood` method of `FoodList` Class.
 
-## Product scope
-### Target user profile
+The add feature implementation is as follows:
+1. New AddCommand object is created by passing in a String containing food details
+2. `splitDetails` is called to split the string
+3. boolean `hasQuantity` is used to record whether the string contains `-q` quantity flag
+4. Split the string using ` -q ` regardless of whether the string contains the quantity part
+5. Check the index of `-n` and `-e` and split the string to save the details of the food name and expiry date
+6. If `hasQuantity` is `true`, return a String array with `name`, `date` and `quantity`, else only return
+a String array with `name` and `date`.
+7. Create a new `Food` object and use `addFood` to add the new food into the food list
+8. Return a `CommandResult` to show the successful message to the user
 
-{Describe the target user profile}
 
-### Value proposition
+**Class Diagram**
+
+![ClassDiagram](images/AddCommandClassDiagram.png)
```

Comment:
Is this an inconsistency among all class diagrams? Should it be (A) abstract for all the Command class instead of (C)?
![image](https://user-images.githubusercontent.com/88440925/228428292-ce11f802-e57f-4494-8ed1-0d1f37a711bd.png)
![image](https://user-images.githubusercontent.com/88440925/228428331-47a86606-5428-4fd3-8f96-13d4b3ff9613.png)